build_scripts/ubuntu_sim.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+## Bash script for setting up a PX4 development environment on Ubuntu LTS (16.04).
+## It can be used for installing simulators (only) or for installing the preconditions for Snapdragon Flight or Raspberry Pi.
+##
+## Installs:
+## - Common dependencies and tools for all targets (including Ninja build system and Qt Creator) 
+## - jMAVSim simulator
+## - Gazebo8 simulator
+## - PX4/Firmware source (to ~/src/Firmware/)
+
+# Ubuntu Config
+sudo apt-get remove modemmanager -y
+
+# Ninja build system
+ninja_dir=$HOME/ninja
+echo "Installing Ninja to: $ninja_dir."
+if [ -d "$ninja_dir" ]
+then
+    echo " Ninja already installed."
+else
+    pushd .
+    mkdir -p $ninja_dir
+    cd $ninja_dir
+    wget https://github.com/martine/ninja/releases/download/v1.6.0/ninja-linux.zip
+    unzip ninja-linux.zip
+    rm ninja-linux.zip
+    exportline="export PATH=$ninja_dir:\$PATH"
+    if grep -Fxq "$exportline" ~/.profile; then echo " Ninja already in path" ; else echo $exportline >> ~/.profile; fi
+    . ~/.profile
+    popd
+fi
+
+# Common Dependencies
+echo "Installing common dependencies"
+sudo add-apt-repository ppa:george-edison55/cmake-3.x -y
+sudo apt-get update -y
+sudo apt-get install python-argparse git-core wget zip python-empy qtcreator cmake build-essential genromfs -y
+# required python packages
+sudo apt-get install python-dev -y
+sudo apt-get install python-pip
+sudo -H pip install pandas jinja2
+
+# jMAVSim simulator
+sudo apt-get install ant openjdk-8-jdk openjdk-8-jre -y
+
+# Gazebo simulator
+echo "Installing Gazebo8"
+sudo apt-get install protobuf-compiler libeigen3-dev libopencv-dev -y
+sudo sh -c 'echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list'
+## Setup keys
+wget http://packages.osrfoundation.org/gazebo.key -O - | sudo apt-key add -
+## Update the debian database:
+sudo apt-get update -y
+## Install Gazebo8
+sudo apt-get install gazebo8 -y
+## For developers (who work on top of Gazebo) one extra package
+sudo apt-get install libgazebo8-dev
+
+
+# Clone PX4/Firmware
+clone_dir=~/src
+echo "Cloning PX4 to: $clone_dir."
+if [ -d "$clone_dir" ]
+then
+    echo " Firmware already cloned."
+else
+    mkdir -p $clone_dir
+    cd $clone_dir
+    git clone https://github.com/PX4/Firmware.git
```

Comment:
@bkueng That isn't needed - the submodules are automatically fetched when you build.